.NET/buildtools/checkSpec.ps1

```diff
@@ -0,0 +1,129 @@
+# Check for empty and duplicate inputs in Specs
+Param ([string]$rootFolder = (get-item $(Get-Location).Path).parent.FullName)
+
+$SpecsFile = Join-Path -Path $rootFolder -ChildPath "Specs"
+$filePattern = "*.json"
+
+$global:totalEmpty = 0
+$global:totalDuplicate = 0
+$global:emptyFileList = New-Object System.Collections.Generic.List[string]
+$global:duplicateFileList = New-Object System.Collections.Hashtable
+
+function SpecInfo()
+{
+    foreach ($file in $input) 
+    {   
+        $parentName = $file.FullName | Split-Path -parent | Split-Path -leaf
+        $typeFolder = $file.FullName | Split-Path -parent | Split-Path -parent | Split-Path -leaf
+		$ret = Get-Content $file.FullName -encoding utf8 | ConvertFrom-Json
+        CheckSpec -spec $ret -type $typeFolder -parent $parentName -name $file.Name
+    }
+
+    Write-Host("Total empty input:`t" + $global:totalEmpty)
+    Write-Host("Total duplicate input:`t" + $global:totalDuplicate)
+
+    if ($global:totalEmpty)
+    {
+        Write-Host("Contain empty input `"`" in these files")
+        foreach ($case in $global:emptyFileList)
+        {
+            Write-Host($case)
```

Comment:
It would be better if for each spec that "fails" there is a count for each duplicate line.
The current output seems a little ambiguous.